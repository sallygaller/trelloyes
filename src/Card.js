// When the "delete card" button is clicked on any card, remove all references to that card from state.
// When the "add random card" button is clicked, generate a random card and add it to the state and the appropriate list.
// In the resources below, we've supplied you with a function for generating a random card.

import React from 'react';
import './Card.css';

export default function Card(props) {
    return (
        <div className='Card'>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
            <button
                onClick={ () => props.onDeleteCard(props.id)}
                type='button'>
                Delete card
            </button>
        </div>
    )
}

Card.propTypes = {
    onClickDelete: () => {}
}