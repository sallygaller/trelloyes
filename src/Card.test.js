import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected when the Card content is long', () => {
    const tree = renderer
        .create(<Card name="Card" content={"Blah blah blah blah blah blah blah blah blah blah blah"} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the UI as expected when the title is blank', () => {
    const tree = renderer
        .create(<Card name="Card" title={""} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});