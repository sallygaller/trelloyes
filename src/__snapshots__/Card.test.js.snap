// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected when the Card content is long 1`] = `
<div
  className="Card"
>
  <h3 />
  <p>
    Blah blah blah blah blah blah blah blah blah blah blah
  </p>
</div>
`;

exports[`renders the UI as expected when the title is blank 1`] = `
<div
  className="Card"
>
  <h3>
    
  </h3>
  <p />
</div>
`;
